android/app/src/main/kotlin/io/lantern/model/BaseModel.kt

```diff
@@ -258,11 +258,17 @@ abstract class BaseModel(
         val subscriberID = args["subscriberID"] as String
         db.unsubscribe(subscriberID)
         activeSubscribers.remove(subscriberID)
+
+        if (activeSubscribers.isEmpty()) {
+            destroy()
+        }
     }
 
     fun destroy() {
         activeSubscribers.forEach {
             db.unsubscribe(it)
         }
+        // Quit the async handler thread
+        asyncHandlerThread.quitSafely()
```

Comment:
This just makes sure the background thread is shut down gracefully when the model is destroyed